cogs/annual_handover_and_reset.py

```diff
@@ -0,0 +1,223 @@
+"""Contains cog classes for annual handover and role reset functionality."""
+
+
+from collections.abc import Sequence
+
+__all__: Sequence[str] = ("CommitteeHandoverCommandCog", "AnnualRolesResetCommandCog")
+
+import logging
+from logging import Logger
+from typing import Final
+
+import discord
+
+from db.core.models import GroupMadeMember
+from utils import AllChannelTypes, CommandChecks, TeXBotApplicationContext, TeXBotBaseCog
+
+logger: Logger = logging.getLogger("TeX-Bot")
+
+
+class CommitteeHandoverCommandCog(TeXBotBaseCog):
+    """Cog class that defines the "/committee-handover" command."""
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="committee-handover",
+        description="Initiates the annual Discord handover procedure for new committee",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def committee_handover(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "committee-handover" command.
```

Comment:
Use function command name in function docsting: `...the "committeee_handover" command.`